### ISSUE TITLE
[feature request] Get/set fastmath CPU bit (and some other FPU flags?)

### ISSUE THREAD

**Issue by vadimkantorov** [pytorch/pytorch#88805]
### 🚀 The feature, motivation and pitch

Originally proposed at: https://github.com/pytorch/pytorch/issues/81690#issuecomment-1309911123

Loading third-party libraries may affect CPU fpu bits such as denormals mode and fastmath (also affecting CPU reproducibility, see https://moyix.blogspot.com/2022/09/someones-been-messing-with-my-subnormals.html). It's useful to be able to detect and reset these bits directly from PyTorch API. Currently there only exists `torch.set_flush_denormal`. There is no `get_flush_denormal`.

It would be useful to have extensive set of functions for getting/setting these CPU flags. It may also be good for having a single function for getting/setting all such flags. Then it'd be easy to dump to a file all flags and then recover them in one go and also provide end-user visibility into this age-old concern.

**Comment by vadimkantorov** [comment]
Or is fastmath the same as flush_denormals? If so and there are no other important bits (maybe controlling default rounding?), then there is no issue...

**Comment by jgong5** [comment]
> Or is fastmath the same as flush_denormals? If so and there are no other important bits (maybe controlling default rounding?), then there is no issue...

Yes. See this: https://github.com/gcc-mirror/gcc/blob/master/libgcc/config/i386/crtfastmath.c
My understanding is that only DAZ and FTZ flags are concerning here and `torch.set_flush_denormal` should work fine. But having a getter function `torch.get_flush_denormal` also makes sense to me.

**Comment by vadimkantorov** [comment]
also, maybe renaming the method to be more explicitly about cpu

**Comment by vadimkantorov** [comment]
And also, a question is what should get_flush_denormal return if DAZ and FTZ flags are set to different values (so conflating them into one bool blag is somewhat leaky?)

**Comment by jgong5** [comment]
> And also, a question is what should get_flush_denormal return if DAZ and FTZ flags are set to different values (so conflating them into one bool blag is somewhat leaky?)

Agree. Also, returning a tuple of flags for DAZ and FTZ sounds overkill (does the user really care about these individual flags?) How about we return False as long as any of them is not set?
